src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -376,7 +376,7 @@ public void Destroy()
 			var cacheable =
 				new List<object>(result.Count + 1)
 				{
-					aliases == null ? ts : new object[] { ts, aliases }
+					aliases == null ? ts : new object[] { ts, aliases.ToArray<object>() }
```

Comment:
`ToArray` call seems not needed. We still can read it as `object[]` after de-serialization without this transformation..

Reply:
[This](https://github.com/nhibernate/nhibernate-core/pull/3184/files#diff-ca8980fb225a4228f73f304c834c6d4793a8d2370627ab0cd99c4bc63e663b30R97) fails without this call, because a `string[]` ends up in the cached value, which is not supported by the JsonCacheSerializer (unless adding a registration for it, but then, that means this fix here is a breaking change).

Reply:
Ok. But it looks like an implementation flaw to me. Who registers `object[]` as supported type? This code should treat any arrays as `object[]` which .NET supports natively.

Reply:
This security feature quite bother me, but since I have not studied what attacks may be possible by relaxing it, I just do with it.

We could also choose to do a new release of JsonCacheSerializer with a few more innocuous types registered like `string[]` or even all basic atomic type arrays. Then tell as a possible breaking change that users of that serializer needs to upgrade to its latest version and check their own registration for potential conflicts with new built-in registrations. But so, that would mean a possible breaking change here in NHibernate and another one in the caches.